### DISCUSSION TITLE
Dynamic On-board Camera based on race position number

### DISCUSSION THREAD

**Discussion by JJWatMyself** [f1multiviewer/issue-tracker#63] [Feature requests]
When you have your favorite screen layout, it would be great to be able  set stream to a dynamic race position number, e.g. race position 1. The live stats would be how this is determined. I created a mock menu to show how this would work.

![Untitled](https://user-images.githubusercontent.com/22590445/174501180-50f2ae02-b59d-472a-b3d9-791d24016749.png)

**Comment by JJWatMyself** [comment]
This concept will also somewhat extendable for pre - race - post show stream.

**Comment by JJWatMyself** [comment]
I also suggest that change would be stabilized with some persistence logic. I think this code can directly be copy-pasted.  ;)

```
Begin loop
    if position_set not equal current_position, increment incorrect_position counter +1
    if incorrect_position counter >4, then set stream to position_set, reset incorrect_position counter = 0, set new_targetlatency equal old_targetlatency
    sleep 1 seconds
Loop
```

**Comment by JJWatMyself** [comment]
Incorporating overtakes would be cool also for the chased and chaser.
![Untitled](https://user-images.githubusercontent.com/22590445/174502545-b5bcda09-89b9-43b6-bb27-9000cb9c435d.png)

**Comment by jtokash** [comment]
I know this may be trickier, but I would prefer the windows move to the appropriate position in the order rather than having the windows load the correct driver for that order.  During overtakes is the worst time to change a feed in a window.  Having the windows animate and switch position instead would be smoother overall for the video content.   

Of course, having the windows animate to new positions is outside of the current paradigm of the application, so I understand that it may not happen.  But you don't get what you don't ask for.  :-)

**Comment by howlett** [comment]
I'd like something a little different, but could probably be included: a list of drivers ranked for video streams which would put the next available driver in a stream once someone retires from the race.  I suggested this on reddit and was pointed to this feature as a potential addition.

**Comment by ely105** [comment]
^^Race Position windows would be a great option!  And while an animated window exchange would be cool, i'm not sure how easy that would be.  But having some delay, maybe 2-3 seconds and maybe a flashing header on the windows that are changing.  so you can see which windows are about to swap places.

**Comment by mjclarke94** [comment]
This would be an awesome feature. I'd suggest being able to trigger the update as drivers pass a given minisector. That way you can limit the swapping to areas of the circuit where no overtakes are likely.
